### PR TITLE
Add mail transporter, reset password and verify email functionality:

### PR DESCRIPTION
**Pull request by Szywacz**
- email-verify-link endpoint for requesting email verification link
- email-verify endpoint for veryfing emails
- password-reset-link endpoint for requesting password reset
- password-reset endpoint for password reseting
- verification check on login

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,5 +1,16 @@
...
+VERIFY_EMAIL_TOKEN_SECRET=
 MONGO_URI=
 
-NODE_ENV=
\ No newline at end of file
+FRONTEND_BASE_URL=http://localhost:3000
+
+APP_NAME=Issue Solver
+NODE_ENV=development
+SALT=
```
Comment (dereniowskid885) [inline .env.example:11]: Salt that is used in bcrypt must be an integer. Value from .env is a string so every use of it need to be parsed to int.

File: controllers/authController.js [hunk 2]
```diff
@@ -147,11 +162,168 @@ const logout = async (req, res) => {
...
+    const appBaseUrl = process.env.FRONTEND_BASE_URL;
+    const resetEndpoint = '/password-reset';
+
+    const resetLink = `${appBaseUrl}${resetEndpoint}/${resetToken}`;
+
+    await transporter
+      .sendMail({
+        to: email,
+        subject: `${process.env.APP_NAME} - Password Reset`,
+        html: `<div>Here is your password reset link: ${resetLink} </div>`,
```
Comment (dereniowskid885) [inline controllers/authController.js]: Please add a resetLink to a href of anchor tag

File: models/User.js [hunk 3]
```diff
@@ -32,13 +32,17 @@ const userSchema = new mongoose.Schema(
...
+      default: false,
+    },
   },
   { timestamps: true },
 );
 
 userSchema.pre('save', async function (next) {
   if (!this.isModified('password')) return next();
-  this.password = await bcrypt.hash(this.password, 12);
+  this.password = await bcrypt.hash(this.password, process.env.SALT);
```
Comment (dereniowskid885) [inline models/User.js]: add parseInt here for process.env.SALT

**Comment by dereniowskid885** [comment]
Currently we don't have a check if new password is the same as current password. Should we cover that case ?

File: controllers/authController.js [hunk 4]
```diff
@@ -17,12 +28,12 @@ const register = async (req, res) => {
       return res.status(400).json({ message: 'Email already registered' });
     }
 
-    const usersCount = await User.countDocuments();
-    const role = usersCount === 0 ? 'admin' : 'user';
+    const role = 'user';
```
Comment (concierginho) [inline controllers/authController.js]:
```suggestion
```

File: controllers/authController.js [hunk 5]
```diff
@@ -17,12 +28,12 @@ const register = async (req, res) => {
...
 
     const user = new User({ email, password, role });
```
Comment (concierginho) [inline controllers/authController.js]:
```suggestion
    const user = new User({ email, password, 'user' });
```

**Pull request merged**